### PR TITLE
Upgrade intellij-related gradle plugins, upgrade to 2022.1

### PR DESCRIPTION
**Pull request by ice1000**
Compiling with the old code runs into an OOM on my computer, but upgrading to 1.6.20 fixes the problem.

I guess the newer versions are getting better :p

### CODE REVIEW

**Comment by ice1000** [comment]
Well, I'm not so lucky. Upgrading Kotlin does allow me to start the compilation (it cannot start compilation for an OOM before), but still cannot finish compiling

**Comment by ice1000** [comment]
Worked!! Yaas

**Comment by ice1000** [comment]
```

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2m 4s
30 actionable tasks: 10 executed, 20 up-to-date
```

Tesla
He compiled the code
Yes
YES
Tesla can use the code

**Comment by ice1000** [comment]
[Arend-1.7.0.2.zip](https://github.com/JetBrains/intellij-arend/files/8423464/Arend-1.7.0.2.zip)
image

**Comment by ice1000** [comment]
[Arend-1.7.0.2.zip](https://github.com/JetBrains/intellij-arend/files/8518452/Arend-1.7.0.2.zip)

**Comment by ice1000** [comment]
221 now has stable release

File: src/main/kotlin/org/arend/actions/ArendSmartSelectProvider.kt [hunk 1]
```diff
@@ -20,6 +20,8 @@ class ArendSmartSelectProvider : SmartSelectProvider<ArendSmartSelectProvider.Co
 
     data class Context(val file: ArendFile, val editor: Editor, val selectionRange: TextRange)
 
+    override fun canIncreaseSelection(source: Context) = true
```
Comment (valis) [inline src/main/kotlin/org/arend/actions/ArendSmartSelectProvider.kt:23]: There is no such method in the base class.

Comment (ice1000) [reply src/main/kotlin/org/arend/actions/ArendSmartSelectProvider.kt:23]: It's 2022.1

**Comment by ice1000** [comment]
@valis https://github.com/JetBrains/intellij-arend/commit/3b9b21b5902afdaf6b118b158239ed6ffa593a0b only upgrades plugins. I think maybe you can cherry pick it. The other two commits require 221 IDEA

**Pull request merged**